lib/daemon_runner/client.rb

```diff
@@ -108,25 +126,60 @@ def parse_task(task)
     end
 
     # @private
+    # @param [Class] instance an instance of the task class
+    # @return [Hash<Symbol, String>] schedule parsed in parts: Schedule type and timing
+    def parse_schedule(instance)
+      valid_types = [:in, :at, :every, :interval, :cron]
+      out = {}
+      task_schedule = if instance.respond_to?(:schedule)
+        instance.send(:schedule)
+      else
+        schedule
+      end
+
+      raise ArgumentError, 'Not enough element in the Array' if task_schedule.length < 2
+      raise ArgumentError, 'Invalid schedule type' unless valid_types.include?(task_schedule[0].to_sym)
+
+      out[:type] = task_schedule[0].to_sym
+      out[:schedule] = task_schedule[1]
+      out
+    end
+
+    # @private
     # @param [Array<String, String, Array>] task to run
     # @return [String] output returned from task
     def run_task(task)
       parsed_task = parse_task(task)
       instance = parsed_task[:instance]
+      schedule = parse_schedule(instance)
       class_name = parsed_task[:class_name]
       method = parsed_task[:method]
       args = parsed_task[:args]
       log_line = "Running #{class_name}.#{method}"
       log_line += "(#{args})" unless args.empty?
       logger.debug log_line
```

Comment:
The logging here was kinda meh but it helped determine what was running and what the output from each task.

Now we log the task name once when it is scheduled and the output every the task is run.  That behavior isn't all that helpful.

```
~/code/cassandra-utils[per-task-scheduling*]% be bin/cass-util stats
D, [2016-10-21T10:55:36.149631 #20962] DEBUG : Running Cassandra::Utils::Stats::Compaction.run!
D, [2016-10-21T10:55:36.149728 #20962] DEBUG : Scheduling job type interval with interval 120
D, [2016-10-21T10:55:37.153411 #20962] DEBUG : Running Cassandra::Utils::Stats::Cleanup.run!
D, [2016-10-21T10:55:37.153535 #20962] DEBUG : Scheduling job type interval with interval 120
D, [2016-10-21T10:57:38.434716 #20962] DEBUG : Got: 0
D, [2016-10-21T10:57:38.838204 #20962] DEBUG : Got: 0
D, [2016-10-21T10:59:40.487985 #20962] DEBUG : Got: 0
D, [2016-10-21T10:59:40.659178 #20962] DEBUG : Got: 0
```